cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -241,42 +244,60 @@ func handleError(response *http.Response, method string, err error) (handler.Pro
 	}
+	if response.StatusCode == http.StatusNotFound {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          fmt.Sprintf("%s:%s", method, err.Error()),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
 
 	return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error during execution : %s", err.Error()),
 		response), nil
 }
 
-func (model *Model) setDataLakeTenant() (dataLakeTenant atlasSDK.DataLakeTenant) {
-	cloudProvider := constants.AWS
-	if model.DataProcessRegion.CloudProvider != nil {
-		cloudProvider = *model.DataProcessRegion.CloudProvider
-	}
-	dataLakeTenant = atlasSDK.DataLakeTenant{
-		Name: model.TenantName,
-		CloudProviderConfig: &atlasSDK.DataLakeCloudProviderConfig{
-			Aws: atlasSDK.DataLakeAWSCloudProviderConfig{
-				TestS3Bucket: *model.CloudProviderConfig.TestS3Bucket,
-				RoleId:       *model.CloudProviderConfig.RoleId,
+func (model *Model) setDataLakeTenant() (dataLakeTenant admin.DataLakeTenant) {
+	dataLakeTenant = admin.DataLakeTenant{
+		Name:    model.TenantName,
+		Storage: model.newDataFederationDataStorage(),
+	}
+
+	cloudProviderConfig := model.CloudProviderConfig
+	if cloudProviderConfig != nil && cloudProviderConfig.TestS3Bucket != nil && cloudProviderConfig.RoleId != nil {
+		dataLakeTenant.CloudProviderConfig = &admin.DataLakeCloudProviderConfig{
+			Aws: admin.DataLakeAWSCloudProviderConfig{
+				TestS3Bucket: *cloudProviderConfig.TestS3Bucket,
+				RoleId:       *cloudProviderConfig.RoleId,
 			},
-		},
-		DataProcessRegion: &atlasSDK.DataLakeDataProcessRegion{
-			Region:        *model.DataProcessRegion.Region,
-			CloudProvider: cloudProvider,
-		},
+		}
+	}
 
-		Storage: model.newDataFederationDataStorage(),
+	dataProcessRegion := model.DataProcessRegion
+	if dataProcessRegion != nil && dataProcessRegion.CloudProvider != nil && dataProcessRegion.Region != nil {
+		dataLakeTenant.DataProcessRegion.CloudProvider = *dataProcessRegion.CloudProvider
+		dataLakeTenant.DataProcessRegion = &admin.DataLakeDataProcessRegion{
+			CloudProvider: *dataProcessRegion.CloudProvider,
+			Region:        *dataProcessRegion.Region,
+		}
 	}
+
+	if dataProcessRegion != nil && dataProcessRegion.Region != nil {
+		dataLakeTenant.DataProcessRegion = &admin.DataLakeDataProcessRegion{
+			CloudProvider: constants.AWS,
+			Region:        *dataProcessRegion.Region,
+		}
+	}
+
 	return dataLakeTenant
 }
 
-func (model *Model) newDataFederationDataStorage() *atlasSDK.DataLakeStorage {
-	return &atlasSDK.DataLakeStorage{
+func (model *Model) newDataFederationDataStorage() *admin.DataLakeStorage {
+	return &admin.DataLakeStorage{
 		Databases: model.newDataFederationDatabase(),
 		Stores:    model.newStores(),
 	}
 }
 
-func (model *Model) newDataFederationDatabase() []atlasSDK.DataLakeDatabaseInstance {
+func (model *Model) newDataFederationDatabase() *[]admin.DataLakeDatabaseInstance {
 	if model.Storage == nil {
```

Comment:
Maybe return empty list instead of nil?  (`[]admin.DataLakeDatabaseInstance{}`)
Related to the SDK upgrade where `nil` pointers will not be dumped in the json. Leading to no updates when Databases might be set to nil? However, might not apply to this resource

Reply:
In this case this is used in create and update operations. I think that if they are not set in the configuration, we want them not to be in the json so that they are not set in the create or update